### ISSUE TITLE
[Syncfusion_Flutter_Datagrid] headerRowHeight visual bug

### ISSUE THREAD

**Issue by Shunt22** [syncfusion/flutter-widgets#1671]
Greetings. 

I encountered an interesting behaivour in SfDataGrid stacked headers.
In my minimal example I have two colums with one header.

The problem lies in `headerRowHeight` propery. (Notice that ` headerGridLinesVisibility: GridLinesVisibility.none`,
When I set `headerRowHeight` property to something round like **101** for example - everything looks fine:
![normal](https://github.com/syncfusion/flutter-widgets/assets/7492436/72e93481-e81a-46b7-8e84-2bbb91634969)

As soon as I set  `headerRowHeight` property to **101.5** ( almost anything with .5-.9  works) a strange white horizontal line appears.
![bug](https://github.com/syncfusion/flutter-widgets/assets/7492436/81af9c6d-1fe3-49bb-8f61-5571f84e80d1)

Tested on Windows, using: 
```
  syncfusion_flutter_datagrid: ^24.2.6
  syncfusion_flutter_core: ^24.2.6
```

```
import 'package:flutter/material.dart';
import 'package:syncfusion_flutter_core/theme.dart';
import 'package:syncfusion_flutter_datagrid/datagrid.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        useMaterial3: true,
      ),
      home: const StackedHeaderDataGrid(),
    );
  }
}

class StackedHeaderDataGrid extends StatelessWidget {
  const StackedHeaderDataGrid({super.key});

  List<GridColumn> _getColumns() {
    List<GridColumn> columns;
    columns = <GridColumn>[
      GridColumn(
          columnName: 'first_name',
          label: Container(
            alignment: Alignment.centerLeft,
            padding: const EdgeInsets.all(8.0),
            child: const Center(
              child: Text(
                'First name',
                overflow: TextOverflow.ellipsis,
              ),
            ),
          )),
      GridColumn(
          columnName: 'last_name',
          label: Container(
            alignment: Alignment.centerRight,
            padding: const EdgeInsets.all(8.0),
            child: const Center(
              child: Text(
                'Last Name',
                overflow: TextOverflow.ellipsis,
              ),
            ),
          )),
    ];
    return columns;
  }

  List<StackedHeaderRow> _getStackedHeaderRows() {
    List<StackedHeaderRow> stackedHeaderRows;
    stackedHeaderRows = <StackedHeaderRow>[
      StackedHeaderRow(cells: <StackedHeaderCell>[
        StackedHeaderCell(
            columnNames: <String>[
              'first_name',
              'last_name',
            ],
            child: const Center(
              child: Text("NAME"),
            )),
      ])
    ];
    return stackedHeaderRows;
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SfDataGridTheme(
        data: SfDataGridThemeData(
          headerColor: Colors.blue,
        ),
        child: SfDataGrid(
          headerGridLinesVisibility: GridLinesVisibility.none,

          /// CHANGE HERE
          headerRowHeight: 101.5,
          //headerRowHeight: 101,

          /// !CHANGE HERE

          columnWidthMode: ColumnWidthMode.fill,
          source: CustomDataGridSource(),
          columns: _getColumns(),
          stackedHeaderRows: _getStackedHeaderRows(),
        ),
      ),
    );
  }
}

class CustomDataGridSource extends DataGridSource {
  List<DataGridRow> data = [
    const DataGridRow(cells: [
      DataGridCell<String>(columnName: "first_name", value: "John"),
      DataGridCell<String>(columnName: "last_name", value: "Smith"),
    ])
  ];

  @override
  List<DataGridRow> get rows => data;

  @override
  DataGridRowAdapter? buildRow(DataGridRow row) {
    return DataGridRowAdapter(
        cells: row
            .getCells()
            .map((e) => Center(child: Text(e.value.toString())))
            .toList());
  }
}
```

**Comment by Tamilarasan-Paranthaman** [comment]
Hi @Shunt22,

We have thoroughly tested the provided sample using the latest versions of DataGrid (24.2.6) and Flutter SDK (3.19.0) on the Windows platform. Regrettably, we were unable to replicate the reported issue on our end. The functionality appears to be working correctly without encountering the problem you have described. For your reference, we have attached a video showcasing the output.

 <details>
 <summary>Output Image</summary>

![image](https://github.com/syncfusion/flutter-widgets/assets/93904422/73eecddf-fa8e-42ab-9620-8fb1b463da55)

</details>

 <details>
 <summary>Output Video</summary>

https://github.com/syncfusion/flutter-widgets/assets/93904422/b3d133bd-9231-4807-ad40-5118fcebfe7c



</details>

Regards,
Tamilarasan

**Comment by Shunt22** [comment]
> Hi @Shunt22,
> 
> We have thoroughly tested the provided sample using the latest versions of DataGrid (24.2.6) and Flutter SDK (3.19.0) on the Windows platform. Regrettably, we were unable to replicate the reported issue on our end. The functionality appears to be working correctly without encountering the problem you have described. For your reference, we have attached a video showcasing the output.

That's interesting! I've upgrade Flutter to 3.19.0, but the issue still exists on my end.
Any ideas what could cause this issue?
Anyway, I've found a workaround by using `headerRowHeight: 101.5.ceilToDouble(),`


Here is my flutter doctor:
```
[√] Flutter (Channel stable, 3.19.0, on Microsoft Windows [Version 10.0.19045.4046], locale ru-RU)
    • Flutter version 3.19.0 on channel stable at D:\Programs\FlutterSDK
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision bae5e49bc2 (6 days ago), 2024-02-13 17:46:18 -0800
    • Engine revision 04817c99c9
    • Dart version 3.3.0
    • DevTools version 2.31.1

[√] Windows Version (Installed version of Windows is version 10 or higher)

[!] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at D:\Programs\AndroidSDK
    • Platform android-33, build-tools 33.0.1
    • Java binary at: D:\Programs\AndroidStudio3.0\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    X Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more details.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop Windows apps (Visual Studio Build Tools 2022 17.8.6)
    • Visual Studio at D:\Programs\VS2022
    • Visual Studio Build Tools 2022 version 17.8.34525.116
    • Windows 10 SDK version 10.0.22621.0

[√] Android Studio (version 2020.3)
    • Android Studio at D:\Programs\AndroidStudio3.0
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] IntelliJ IDEA Community Edition (version 2020.2)
    • IntelliJ at D:\Programs\IntelliJ IDEA Community Edition 2020.2.3
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19045.4046]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 121.0.6167.185
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 121.0.2277.112

[√] Network resources
    • All expected network resources are available.
```

**Comment by abineshPalanisamy** [comment]
Hi @Shunt22 ,

Based on the information provided, we thoroughly tested the provided sample using the latest versions of DataGrid (25.1.35) and Flutter SDK (3.19.3). Regrettably, we were unable to replicate the reported issue on our end when the headerGridLinesVisibility was set to none. Therefore, we checked the sample with headerGridLinesVisibility and also set a different color for the grid lines. We found that the color of the grid lines appeared as expected. We have attached the sample file and output images for your reference. If this does not meet your requirements, kindly modify the attached sample to reproduce the issue.

- **Sample :** 
  
  [SfDataGrid.zip](https://github.com/syncfusion/flutter-widgets/files/14720260/SfDataGrid.zip)

- **Output images :** 

    [Output images.zip](https://github.com/syncfusion/flutter-widgets/files/14720345/Output.images.zip)

 

Regards,
Abinesh P
